### ISSUE TITLE
Fetch notes for completed tasks

### ISSUE THREAD

**Issue by xavdid** [wunderlist/api#66]
Currently, it doesn't seem like there's a way to get all the notes for completed tasks. When you hit `/notes` with a list id, it gives an array of notes for incomplete tasks. Information about a completed task can be found only on the direct id lookup.

Is this expected behavior? Would it be possible to add a bool to that end point that returns notes for all completed or incomplete tasks, like the `/tasks` endpoint itself? 

Thanks!

**Comment by vsmart** [comment]
Hi @xavdid, yes this is expected behaviour. To get all notes only for completed tasks, you'd need to filter out those notes with task-ids from completed tasks.

**Comment by xavdid** [comment]
To clarify, is it possible to fetch all completed task notes for a given list id with a single call?

**Comment by vsmart** [comment]
@xavdid No. You can fetch *all* notes for a given list_id, but you'll also need to fetch all tasks to get the task-ids and filter out which notes are from non-completed tasks.

**Comment by xavdid** [comment]
Maybe I'm not being clear- `curl`ing `https://a.wunderlist.com/api/v1/notes?list_id=<ID>` returns notes only for incomplete tasks. 

As soon as I complete a task, its note ceases to be returned by that call. If I uncomplete the task, the note shows up again

**Comment by vsmart** [comment]
Ah, I see what you are saying. Sorry I misunderstood earlier @xavdid. The assumption is that a client will only need to display uncompleted tasks. If you do need to fetch a particular completed task's note, you'll have to fetch it by id as you suspected. Does that make sense? @xavdid

**Comment by xavdid** [comment]
Yep, that's what I figured. thanks!

~DavidB
[davidbrownman.com]
c: 303.819.0850

**Comment by shyba** [comment]
Hi @vsmart, sorry to bother on a closed issue.
I just had the same issue and I would like to avoid flooding your server with requests for each completed Task.
My use case is the following: I am making an open source application to do personal finance based on a list my family keeps shared. Every completed task has a note with the value that we spent on this bill and each Task is a paid bill. So, I would like to fetch all notes from completed tasks in this list to make graphics and help to control budget. After it is done for us, I will generalize the application and blog about this use case.
We tried several apps, but my family is composed also by non-tech people and Wunderlist was the only one that nobody complained. So I am implementing this app to get the missing features for a personal finance app.
I just started, but [feel free to browse it](https://github.com/shyba/wunder-finance).
